File: Algorithms/src/2161. Partition Array According to Given Pivot/Code/solution.go
func pivotArray(nums []int, pivot int) []int {

    // Store elements smaller than pivot
    smaller := []int{}

    // Store elements equal to pivot
    equal := []int{}

    // Store elements greater than pivot
    greater := []int{}

    // Classify every element
    for _, num := range nums {
        if num < pivot {
            smaller = append(smaller, num)
        } else if num == pivot {
            equal = append(equal, num)
        } else {
            greater = append(greater, num)
        }
    }

    // Build final answer
    result := []int{}

    result = append(result, smaller...)
    result = append(result, equal...)
    result = append(result, greater...)

    return result
}

File: Algorithms/src/2161. Partition Array According to Given Pivot/Code/solution.cpp
class Solution
{
public:
    vector<int> pivotArray(vector<int> &nums, int pivot)
    {
        // Store elements smaller than pivot
        vector<int> smaller;

        // Store elements equal to pivot
        vector<int> equal;

        // Store elements greater than pivot
        vector<int> greater;

        // Classify every element into one of three groups
        for (int num : nums)
        {
            if (num < pivot)
            {
                smaller.push_back(num);
            }
            else if (num == pivot)
            {
                equal.push_back(num);
            }
            else
            {
                greater.push_back(num);
            }
        }

        // Append equal elements after smaller elements
        smaller.insert(smaller.end(), equal.begin(), equal.end());

        // Append greater elements at the end
        smaller.insert(smaller.end(), greater.begin(), greater.end());

        // Return the final partitioned array
        return smaller;
    }
};